objective_epoch.py

```diff
@@ -0,0 +1,220 @@
+from mlmicrophysics.models import DenseNeuralNetwork
+from mlmicrophysics.data import subset_data_files_by_date, assemble_data_files
+import numpy as np
+import pandas as pd
+from sklearn.preprocessing import StandardScaler, MinMaxScaler, MaxAbsScaler, RobustScaler, OneHotEncoder
+from sklearn.metrics import confusion_matrix, accuracy_score, mean_absolute_error, mean_squared_error
+from mlmicrophysics.metrics import heidke_skill_score, peirce_skill_score, hellinger_distance, root_mean_squared_error, r2_corr
+import argparse
+import yaml
+from os.path import join, exists
+import os
+from datetime import datetime
+import logging
+from keras.losses import huber_loss
+
+import optuna
+from aimlutils.hyper_opt.utils import trial_suggest_loader
+from aimlutils.hyper_opt.base_objective import *
+
+
+logger = logging.getLogger(__name__)
+
+scalers = {"MinMaxScaler": MinMaxScaler,
+           "MaxAbsScaler": MaxAbsScaler,
+           "StandardScaler": StandardScaler,
+           "RobustScaler": RobustScaler}
+
+class_metrics = {"accuracy": accuracy_score,
+                 "heidke": heidke_skill_score,
+                 "peirce": peirce_skill_score}
+
+reg_metrics = {"rmse": root_mean_squared_error,
+               "mae": mean_absolute_error,
+               "r2": r2_corr,
+               "hellinger": hellinger_distance,
+               "mse": mean_squared_error,
+               "huber": huber_loss}
+
+def leaky(x):
+    return tf.nn.leaky_relu(x, alpha=0.01)
+
+def ranked_probability_score(y_true_discrete, y_pred_discrete):   
+    y_pred_cumulative = np.cumsum(y_pred_discrete)
+    y_true_cumulative = np.cumsum(y_true_discrete)
+    return np.mean((y_pred_cumulative - y_true_cumulative) ** 2) / float(y_pred_discrete.shape[1] - 1)
+
+def objective(trial, config):
+
+    # Get list of hyperparameters from the config
+    hyperparameters = config["optuna"]["parameters"]
+
+    # Now update some hyperparameters via custom rules
+    class_activation = trial_suggest_loader(trial, hyperparameters["class_activation"])
```

Comment:
For future code succinctness and scalability, it would be good to replace the series fo trial_suggest_loader calls with a for loop. For example:
```python
trial_hyperparameters = {}
for param_name in hyperparameters.keys():
    trial_hyperparameters[param_name] = trial_suggest_loader(trial, hyperparameters[param_name])
```

Reply:
updated!